src/server/pfs/server/master.go

```diff
@@ -334,82 +368,115 @@ func (d *driver) runCronTrigger(ctx context.Context, branch *pfs.Branch) error {
 	}
 }
 
-func (d *driver) finishRepoCommits(ctx context.Context, repoKey string) error {
-	return errors.EnsureStack(d.commits.ReadOnly(ctx).WatchByIndexF(pfsdb.CommitsRepoIndex, repoKey, func(ev *watch.Event) error {
-		if ev.Type == watch.EventError {
-			return ev.Err
+func (d *driver) finishRepoCommits(ctx context.Context, repoPair pfsdb.RepoPair) error {
+	chanName := fmt.Sprintf("%s%d", pfsdb.CommitsRepoChannelName, repoPair.ID)
```

Comment:
We should have a function for this in pfsdb.  This is very magical and far away from the schema here.